packages/wallets/bitkeep/src/adapter.ts

```diff
@@ -9,10 +10,185 @@ import {
+    readonly supportedTransactionVersions = null;
+
+    private _connecting: boolean;
+    private _wallet: SolanaProvider | null;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' || typeof document === 'undefined'
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.NotDetected;
+
+    constructor(config: BitgetWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            scopePollingDetectionStrategy(() => {
+                if (window.bitkeep?.solana?.isBitKeep) {
+                    this._readyState = WalletReadyState.Installed;
+                    this.emit('readyStateChange', this._readyState);
+                    return true;
+                }
+                return false;
+            });
+        }
+    }
+
+    get publicKey() {
+        return this._publicKey;
+    }
+
+    get connecting() {
+        return this._connecting;
+    }
+
+    get readyState() {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Installed) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            let adapter: typeof SolanaProvider;
+            try {
+                adapter = (await import('@bitget-wallet/web3-sdk')).SolanaAdapter;
+            } catch (error: any) {
+                throw new WalletLoadError(error?.message, error);
+            }
+
+            let wallet;
+            try {
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                wallet = window.bitkeep!.solana! || new adapter?.getProvider();
+            } catch (error: any) {
+                throw new WalletConfigError(error?.message, error);
+            }
```

Comment:
There are a few issues here --

First, this code will never be reached unless `window.bitkeep.solana.isBitKeep` is true, because the `readyState` will not be "installed". Check out the Solflare adapter for a wallet that can both be injected and loaded with an SDK for how to handle this.

Second, you probably don't want to load the adapter SDK if the wallet is already installed, right? It's only used in the event that it isn't installed.